client/actions/locations.js

```diff
@@ -1,5 +1,112 @@
+
+
+const locationsSearch = (searchQuery) => (dispatch) => {
+    dispatch(
+        {
+            type: 'LOCATIONS_CHANGE_LIST_SETTINGS',
+            payload: {
+                searchQuery,
+            },
+        });
+};
+
+const editLocation = (location) => (dispatch) => {
+    dispatch({type: 'EDIT_LOCATION',
+        payload: {
+            location,
+        }});
+};
+
+
+const updateLocation = (original, updated) => (
+    (dispatch, getState, {api, notify}) => {
+        if (original) {
+            return api('locations').save(original, updated)
+                .then((location) => {
+                    dispatch({type: 'UPDATE_LOCATION', payload: location}),
+                    notify.success(gettext('The location has been updated'));
+                },
+                (error) => {
+                    notify.error(getErrorMessage(error, gettext('Failed to updated location!')));
+                    return Promise.reject(error);
+                });
+        } else {
+            return api('locations').save({}, updated)
+                .then((newloctaion) => {
+                    dispatch(self.clearEdits());
+                    dispatch(self.searchLocations(getState().locations.searchQuery));
+                    notify.success(gettext('The location has been created'));
+                    return Promise.resolve(newloctaion);
+                },
+                (error) => {
+                    let errorMessage = get(error, 'data._issues.unique_name.unique') ?
+                        gettext('A location matching this one already exists') : getErrorMessage(error);
+
+                    notify.error(errorMessage);
+                });
+        }
+    }
+);
+
+const createLocation = () => (
+    (dispatch) => {
+        dispatch({
+            type: 'CREATE_LOCATION',
+            payload: '',
+        });
+    }
+);
+
+
+const deleteLocationConfirmation = (location) => (
+    (dispatch, getState, {api}) => {
+        dispatch(showModal({
+            modalType: MODALS.CONFIRMATION,
+            modalProps: {
+                body: gettext(`Do you want to delete the location \"${location.name}\"?`),
+                action: () => dispatch(deleteLocation(location)),
+                autoClose: true,
+            },
+        }));
+    }
+);
+
+const deleteLocation = (location) => (
+    (dispatch, getState, {api}) => {
+        api('locations').remove(location)
+            .then(
+                dispatch({type: 'REMOVE_LOCATION', payload: {location}}));
+    }
+);
+
+const clearEdits = () => (
+    (dispatch) => {
+        dispatch({
+            type: 'CANCEL_EDIT',
+            payload: '',
+        });
+    }
+);
+
+const getMoreLocations = () => (
+    (dispatch, getState) => {
+        dispatch({
+            type: 'START_LOADING_LOCATIONS',
+            payload: '',
+        }),
+        dispatch(self.getLocation(getState().locations.searchQuery, false, getState().locations.pageNum + 1))
```

Comment:
Use `getLocationSearchQuery` selector.

Also, create a selector that will return the page number, and use that selector here. (Always use selectors when getting data from the store. If the store schema changes you only change the selector instead)

Reply:
Done!